File: Estruturas de dados lineares/java/Fila.java
class FIFO {
    //dados
    int[] array;
    int n;

    //construtor
    public FIFO() {
        this(100);
    }

    /**
     * 
     * @param tamanho
     */
    public FIFO(int tamanho) {
        this.array = new int[tamanho];
        n = 0;
    }

    /**
     * 
     * @param elemento
     * @throws Exception
     */
    public void inserir(int elemento) throws Exception {
        if ( n == array.length ) {
            throw new Exception("array cheio");
        }

        array[n] = elemento;
        n++;
    }

    /**
     * 
     * @return
     * @throws Exception
     */ 
    public int remover() throws Exception{
        if ( n == 0 ) {
            throw new Exception("array vazio");
        }

        int resposta = array[0];
        n--;

        for(int i = 0; i < n ; i++) {
            array[i] = array[i + 1];
        }

        return resposta;
    }

    public void mostrar() {
        System.out.print("[");
        for(int i = 0; i < n; i++ ) {
            System.out.print(" "+ array[i]);
        }

        System.out.println(" ]");
    }
}



public class Fila {
    public static void main(String[] args) throws Exception {
        //dados
        FIFO fila = new FIFO(5);

        //exemplos de comandos
        /*fila.inserir(0);
        fila.inserir(1);
        fila.inserir(2);
        fila.inserir(3);
        fila.inserir(4);
        fila.mostrar();
        fila.remover();
        fila.remover();
        fila.mostrar();*/
    }
}

File: Estruturas de dados lineares/C/Fila.c
#include "stdio.h"
#include "stdlib.h"

#define MAXTAM 6

typedef struct {
    //dados
    int array[MAXTAM];
    int n;
}Fila;

//funções
void iniciar(Fila *fila);
void inserir(int elemento, Fila *fila);
int remover(Fila *fila);
void mostrar(Fila *fila);


int main(void) {
    //dados
    Fila fila;

    //exemplos de comando
    inserir(1, &fila);
    inserir(5, &fila);
    remover(&fila);
    inserir(6, &fila);
    remover(&fila);
    mostrar(&fila);
}

/*
    @param fila
*/
void iniciar(Fila *fila) {
    fila->n = 0;
}

/*
    @param elemento
    @param fila
*/
void inserir(int elemento, Fila *fila) {
    if ( fila->n == MAXTAM ) {
        exit(1);
    }

    fila->array[fila->n] = elemento;

    fila->n++;
}

/*
    @param fila
*/
int remover(Fila *fila) {
    if ( fila->n == 0 ) {
        exit(1);
    }

    int resposta = fila->array[0];
    fila->n--;

    for (int i = 0; i < fila->n; i++ ) {
        fila->array[i] = fila->array[i+1];
    }

    return resposta;
}

/*
    @param fila
*/
void mostrar(Fila *fila) {
    printf("%s", "[ ");

    for(int i = 0; i < fila -> n; i++) {
        printf("%i %s", fila -> array[i], "");
    }

    printf("%s", "]\n");
}

